### PR TITLE
update GitHub actions

### PR DESCRIPTION
**Pull request by cenit**
to avoid deprecated set-env usage

### CODE REVIEW

**Comment by cenit** [comment]
There is a minor cmake issue in CMakeLists.txt, probably due to CMake 3.19, which is still unfixed. I am investigating it now

**Comment by AlexeyAB** [comment]
@cenit Hi, 
Did you find a solution to the problem in the CI tests?

**Comment by cenit** [comment]
Nope, still not found. 
I saw also that travis is in bad shape... should we investigate it too or shall we just discard it? Please tell me what do you prefer

**Comment by AlexeyAB** [comment]
I think we just should discard only some configuration with problems in TravisCI:

image

**Comment by AlexeyAB** [comment]
@cenit Hi,

Also can we disable `AVX` in these 2 lines if we compile it on ARM CPU, like Jetson Nano or new Mac-PC? https://github.com/AlexeyAB/darknet/blob/9db8609e7a07409fe31bf484fb08db385350a5f9/CMakeLists.txt#L175-L176

It is related to this issue: https://github.com/AlexeyAB/darknet/issues/6466

File: .github/workflows/ccpp.yml [hunk 1]
```diff
@@ -80,18 +80,34 @@ jobs:
 
     - uses: lukka/get-cmake@latest
 
-    - name: Restore from cache and run vcpkg
+    - name: 'Install CUDA'
```
Comment (Jmennius) [inline .github/workflows/ccpp.yml:83]: All other jobs that are CUDA-enabled have `-cuda` suffix - so I think it's best to also suffix this one with `-cuda` (i.e. `ubuntu-vcpkg-cuda`).

**Comment by Jmennius** [comment]
@cenit I see that commit b7b6de5 does some things that PR #7164 does - can you consider that PR instead of the commit?

**Comment by cenit** [comment]
> @cenit I see that commit [b7b6de5](https://github.com/AlexeyAB/darknet/commit/b7b6de55a91e2e28ca94f85ccec9a2833be4d4e7) does some things that PR #7164 does - can you consider that PR instead of the commit?

we can merge your PR before this one

**Comment by cenit** [comment]
@AlexeyAB the bug is fixed and the PR is almost ready. I still need to check if the library works ok on downstream projects.
Also, please consider pr #7164 before merging this one. It contains useful ideas, which I also implemented here. The ideas there are not complete, but here there are the fixes to finish them after that PR. To correctly recognize his work I think it might be appropriate

**Comment by stephanecharette** [comment]
In regards to:  "remove cygwin and ubuntu-vcpkg without cuda configs"

...does this mean vcpkg cannot be used anymore to build the CPU version of Darknet?  That would be unfortunate.  Many people (myself included) build it in a VirtualBox VM which doesn't have access to the GPU, so the only way to build and run darknet is the CPU-only mode.

**Comment by AlexeyAB** [comment]
@cenit Hi, Thanks!

> @AlexeyAB the bug is fixed and the PR is almost ready. 

Great!

> I still need to check if the library works ok on downstream projects.

So should I wait for this?

> Also, please consider pr #7164 before merging this one. It contains useful ideas, which I also implemented here. The ideas there are not complete, but here there are the fixes to finish them after that PR. To correctly recognize his work I think it might be appropriate

Should I merge pr #7164 first, and then your pr?


Also is seems there is some issue in CI for this PR `
Darknet Continuous Integration / win-vcpkg-cuda (pull_request) Failing after 78m — win-vcpkg-cuda
`:
image

**Comment by cenit** [comment]
> In regards to:  "remove cygwin and ubuntu-vcpkg without cuda configs"
> 
> 
> 
> ...does this mean vcpkg cannot be used anymore to build the CPU version of Darknet?  That would be unfortunate.  Many people (myself included) build it in a VirtualBox VM which doesn't have access to the GPU, so the only way to build and run darknet is the CPU-only mode.

Absolutely not.
You can find many many configs in cpu mode only. Just CUDA is much more prominent now and to avoid it you have to explicitly opt-out of it

**Comment by cenit** [comment]
> @cenit Hi, Thanks!
> 
> 
> 
> > @AlexeyAB the bug is fixed and the PR is almost ready. 
> 
> 
> 
> Great!
> 
> 
> 
> > I still need to check if the library works ok on downstream projects.
> 
> 
> 
> So should I wait for this?
> 
> 
> 
> > Also, please consider pr #7164 before merging this one. It contains useful ideas, which I also implemented here. The ideas there are not complete, but here there are the fixes to finish them after that PR. To correctly recognize his work I think it might be appropriate
> 
> 
> 
> Should I merge pr #7164 first, and then your pr?
> 
> 
> 
> 
> 
> Also is seems there is some issue in CI for this PR `
> 
> Darknet Continuous Integration / win-vcpkg-cuda (pull_request) Failing after 78m — win-vcpkg-cuda
> 
> `:
> 
> image
> 
> 

Yea please, let me just finish these final problems :)

**Comment by AlexeyAB** [comment]
@cenit Hi,

Was this issue solved?
Or can we merge this PR and expect the rest of the fixes in another PR?

**Comment by cenit** [comment]
win-vcpkg-cuda is still broken. I cannot reproduce the issue but I am working on a big update on vcpkg which might unblock issue here
You can merge this one in the meantime if you prefer, then with another PR i can finish the work on that configuration

**Comment by AlexeyAB** [comment]
@cenit 

Ok, so I will first merge this PR https://github.com/AlexeyAB/darknet/pull/7095 and then merge your PR https://github.com/AlexeyAB/darknet/pull/7095 and then will wait for new another PR.

**Comment by Jmennius** [comment]
> @cenit
> 
> Ok, so I will first merge this PR #7095 and then merge your PR #7095 and then will wait for new another PR.

Did you mean #7164 as the first one?

**Comment by AlexeyAB** [comment]
@Jmennius Yes, sure )

**Comment by cenit** [comment]
>Ok, so I will first merge this PR #7095 and then merge your PR #7095 and then will wait for new another PR.

I can also merge this one if it's ok for you and any conflict arises after you merge #7164

**Comment by AlexeyAB** [comment]
@cenit Yes, please merge it.

**Comment by cenit** [comment]
ok I did it. Merged #7164 then fixed conflicts here and then merged this one
I will monitor events in the near future to see if any big problem arises, then I will try to fix the win-vcpkg-cuda ci

**Comment by xiaoxiaopeng1998** [comment]
@cenit I might meet the error.
My ubuntu version 20.04.
CUDA 10.1.

```
[cmake] CMake Error at CMakeLists.txt:91 (message):
[cmake]   CUDA not found, please build explicitly with -DENABLE_CUDA=OFF if you do
[cmake]   not want CUDA
```
**I have successfully installed the CUDA and cudnn.**
图片

**And I have run it successfully with the command :`./darknet demo`**

So if I change the code and want to recompile it.the only way is `make clean` and `make` in ubuntu?

**Pull request merged**